src/test/java/com/philips/research/bombase/core/nuget/domain/NugetClientTest.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2020-2021, Koninklijke Philips N.V., https://www.philips.com
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.philips.research.bombase.core.nuget.domain;
+
+import com.github.packageurl.MalformedPackageURLException;
+import com.github.packageurl.PackageURL;
+import com.philips.research.bombase.core.meta.PackageMetadata;
+import com.philips.research.bombase.core.nuget.NugetException;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class NugetClientTest {
+    private static final int PORT = 1081;
+    private static final PackageURL PURL = createPurl("pkg:nuget/name@version");
+    private static final String TITLE = "Title";
+    private static final String HOMEPAGE = "https://example.com/home-page";
+    private static final String DESCRIPTION = "Description";
+    private static final String AUTHORS = "Attribution1, Attribution2";
+    private static final String SOURCE_LOCATION = "https://example.com/source";
+    private static final String DOWNLOAD_LOCATION = "https://example.com/binary";
+    private static final String SHA512 = "Sha512";
+    private static final String CATALOG_ENTRY = "http://localhost:" + PORT +
+            "/catalog0/data/timestamp/package.version.json";
+    private static final String LICENSE_EXPRESSION = "MIT";
+    private static final String LICENSE_URL = "https://example.com/mit-license";
+
+    private final NugetClient client = new NugetClient(URI.create("http://localhost:" + PORT));
+    private final MockWebServer mockServer = new MockWebServer();
+
+    static PackageURL createPurl(String purl) {
+        try {
+            return new PackageURL(purl);
+        } catch (MalformedPackageURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @BeforeEach
+    void setUp() throws IOException {
+        mockServer.start(PORT);
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        mockServer.shutdown();
+    }
+
+    @Nested
+    class UnExpectedResults {
+        @Test
+        void skipsUndefinedPackage() {
+            mockServer.enqueue(new MockResponse().setResponseCode(404));
+
+            assertThat(client.getPackageMetadata(PURL)).isEmpty();
+        }
+
+        @Test
+        void withoutCatalogEntry() throws Exception {
```

Comment:
So a good name would be(according to your convention): ``withoutCatalogEntry_returnEmpty``. 
Not sure if I find this naming convention helpful. In Javascript I always was a fan of storytelling in my tests...  see [this](https://github.com/philips-labs/repolinter/blob/master/tests/fixes/github_issue_create_tests.js#L143-L171) example.

Reply:
I try to use a naming convention inspired by Google, where every test title expresses a behaviour of the thing under test. Optionally I add an underscore to visually group tests with the same behaviour for different reasons.

This would lead to e.g.: `noMetadata_noCatalogEntry` (if the behaviour is to not return metadata for this case).